runtime/lua/vim/treesitter/query.lua

```diff
@@ -4,6 +4,59 @@ local memoize = vim.func._memoize
 
 local M = {}
 
+local function is_directive(name)
+  return string.sub(name, -1) == '!'
+end
+
+---@nodoc
+---@class vim.treesitter.query.ProcessedPred
```

Comment:
This is a problem because we need to refer to a neovim "predicate" and tree-sitter "predicate" in the same place, and I can't rename either. What should the names be then? Here's an example of this:

https://github.com/neovim/neovim/blob/6c69b0ba7976e8767c0f2d07ac4fb8a15eea4625/runtime/lua/vim/treesitter/query.lua#L805-L808

805 refers to this processed neovim predicate, while 808 refers to a treesitter predicate

Reply:
We are indeed not 100% consistent (esp. in metadata and optional fields), but this is out of scope here. (We _do_ have `add_directive()` though!)

So here's my suggested convention for now:
1. Keep the distinction between directives (`#foo!`) and predicates (`#bar?`) wherever it exists.
2. If there is a need for a term covering both, use "predicate". (I know that's not ideal, but it's not a regression, and that is all I care about in the context of this PR.)
3. In code, `pred` is fine as a shorthand for both -- only exposed API and docs are critical.

(Treesitter docs are still a bit of a mess and require a general cleanup once we're happier with the general shape of things and have a clearer view of what we _want_ to document, and how.)

Reply:
> (We do have add_directive() though!).

I did a poor job explaining what I was referring to. I meant that docs for `add_predicate()` refer to neovim predicates and docs for `directive_handlers` refer to treesitter predicates. Updated my comment.

As for the convention, i think the PR adheres to all three points.

Reply:
Yep, just noting down for reference; no objections to your PR.